README.rst

```diff
@@ -1,81 +1,98 @@
+For a list of available commands, run ``west -h``. Get help on a
+command with ``west <command> -h``. For example::
 
-Then use the wrapper script to initialize a Zephyr installation with::
+  $ west -h
+  usage: west [-h] [-z ZEPHYR_BASE] [-v]
+              {build,flash,debug,debugserver,attach,list-projects,fetch,pull,rebase,branch,checkout,diff,status,forall}
+              ...
+  [snip]
+  $ west flash -h
+  usage: west flash [-h] [-H] [-d BUILD_DIR] ...
+  [snip]
 
-  $ west init your/zephyr/install-dir
-  $ cd your/zephyr/install-dir
-  $ west <command>
+Test Suite
+----------
 
-The ``west init`` call will:
+To run the test suite, run this from the West repository::
 
-- create your/zephyr/install-dir
-- clone the Zephyr manifest repository (whose URL can be overridden
-  with the ``-u`` option, and branch with ``--mr`` / ``--manifest-rev``)
-- clone the latest West repository (the URL override is ``-w``, and
-  revision/branch override is ``--wr`` / ``--west-rev``)
+  pip3 install -r tests_requirements.txt
+  python3 setup.py test
 
-Running ``west <command>`` from :file:`your/zephyr/install-dir` or
-underneath it will invoke west in "wrapper" mode: any commands other
-than ``init`` will be delegated to the West tree pulled by ``west
-init``.
+Hacking on West
+---------------
 
-This arrangement may seem familiar to Android (platform, not app)
-developers. The source management features of West were indeed
-inspired by the Android Repo tool's features, but West makes
-significant departures from Repo's behavior.
+West is distributed as two Python packages:
 
-Alternative Usage
------------------
+1. A ``bootstrap`` package, which is distributed via PyPI. Running
+   ``pip3 install west`` installs this **bootstrapper package only**.
+2. The "main" ``west`` package, which is fetched by the bootstrapper
+   when ``west init`` is run.
 
-If you don't want to change your system outside of cloning this
-repository, you can also clone West's Git repository and run the
-package as a module:
+This somewhat unusual arrangement is because:
 
-$ python3 -m west
+- One of West's jobs is to manage a Zephyr installation's Git
+  repositories, including its own.
+- It allows easy customization of the version of West that's shipped
+  with non-upstream distributions of Zephyr.
+- West is experimental and is not stable. Users need to stay in sync
+  with upstream, and this allows West to automatically update itself.
 
-Only do this if you know what you're doing (and make sure to install
-the requirements as specified by the install_requires line in
-setup.py).
+To initialize West from a non-default location::
 
-Test Suite
-----------
+  west init -w https://example.com/your-west-repository.git
+
+You can also add ``--west-rev some-branch`` to use ``some-branch``
+instead of ``master``.
+
+To use another manifest repository (optionally with ``--mr
+some-manifest-branch``)::
+
+  west init -u https://example.com/your-manifest-repository.git
```

Comment:
This made me realize that a common usecase would be to specify both a non-default west location and also a non-default manifest repo. Maybe we could add a combined option like:
`west init -uw https://github.com/myuser`, and west would automatically append `/west.git` and `/manifest.git` to the URL in order to fetch those.

Reply:
Eew :).

Sure, we could do something like that. Maybe `-b` or `--url-base`? I'm assuming this is post-0.1.0.

Reply:
>Eew :)
You mean eew `-uw`? `-b` is fine with me :)

Reply:
Oh, just seems a bit weird to me personally, but I see the value for short command lines.

Reply:
I think it makes sense, but this can wait if you don't want to add it now.